### PR TITLE
[14.0][ADD] partner_tag_smart_assignation

### PR DESCRIPTION
**Pull request by ecino**
New module for tagging partners automatically based on search filters or SQL queries.

### CODE REVIEW

**Comment by ecino** [comment]
Anybody willing to review this new module? I think it brings a nice feature addition.

**Review by NL66278 — APPROVED** [review]
:+1: LGTM

**Comment by NL66278** [comment]
@ecino please rebase and force push again to rerun tests

File: partner_tag_smart_assignation/cron/update_cron.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<odoo>
```
Comment (NL66278) [inline partner_tag_smart_assignation/cron/update_cron.xml:1]: On cron data added this way it is best to use the noupdate flag, so changed to the cronfiguration (runtime, active or not, etc.) are not undone by an update.

File: partner_tag_smart_assignation/tests/test_smart_tagger.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    def test_check_validity_dates(self):
+        """
+        Test if the valid_until functionality works correctly
+        """
+
+        # Create a new tag with a 'valid_until' date set to yesterday
+        yesterday = fields.Date.to_string(fields.Date.today() - timedelta(days=1))
+
+        expired_tag = self.create_tag()
+        expired_tag.update({"valid_until": yesterday})
```
Comment (NL66278) [inline partner_tag_smart_assignation/tests/test_smart_tagger.py:138]: You must use `write` here and then call the _check_validity_dates method manually, as the scheduler will not run immediately. `update`  only updates the in memory record.

File: partner_tag_smart_assignation/tests/test_smart_tagger.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        expired_tag.update_partner_tags()
+
+        # Reload the tags from the database
```
Comment (NL66278) [inline partner_tag_smart_assignation/tests/test_smart_tagger.py:141]: This is not needed. Actually updating the database through write, will update the database (and the cache). invalidate_cache can be useful after changing stuff through SQL.

File: partner_tag_smart_assignation/models/hr_department.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
```
Comment (NL66278) [inline partner_tag_smart_assignation/models/hr_department.py:1]: Your module could be useful for many organisations that do not have `hr`  installed. So better not to have `hr`  as a dependency. Put the `hr` modification in a separate module (partner_tag_smart_assignation_hr), that depends on both this module and hr.

**Review by NL66278 — CHANGES_REQUESTED** [review]
:+1: Thanks for your contribution. Please take up requested changed so we can approve this.

**Pull request closed without merging**